date_range/tests/test_date_range.py

```diff
@@ -109,17 +110,10 @@ def test_domain(self):
              ('my_field', '<=', datetime.date(2015, 12, 31))])
 
     def test_date_range_multicompany_1(self):
-        dr = self.date_range.new({
-            'name': 'FS2016',
-            'date_start': '2015-01-01',
-            'date_end': '2016-12-31',
-            'type_id': self.typeB.id,
-            'company_id': self.company.id,
-        })
-        dr._cache.update(dr._convert_to_cache(
-            {'company_id': self.company_2.id}, update=True))
-        dr._onchange_company_id()
-        self.assertFalse(dr.type_id)
+        dr_form = Form(self.date_range)
+        dr_form.type_id = self.typeB
+        dr_form.company_id = self.company_2
+        self.assertFalse(dr_form.type_id)
```

Comment:
@mreficent The `type_id` fiels is now a computed field. We don't need to use a Form to test if the field is set to false when a company change...

Reply:
I think the Form is used here to check that the old onchange functionality still works (onchange method is now replaced by the compute method). Makes sense as a regression test.

Reply:
IMO, it's not our responsibility to check that the functionalities provided by the ORM at core level works as expected. but I'll not block for this.